packages/core/src/client.ts

```diff
@@ -204,11 +207,15 @@ export class Client<
 
   async destroy() {
     if (this.connector) await this.connector.disconnect?.()
+    this.#autoConnecting = false
     this.clearState()
     this.store.destroy()
   }
 
   async autoConnect() {
+    if (this.#autoConnecting) return
+    this.#autoConnecting = true
```

Comment:
Is `#autoConnecting` for making sure this function doesn't run multiple times simultaneously?

Can we change `#autoConnecting` to `#isAutoConnecting` too.

Reply:
Random aside about `autoConnect`: Was thinking we could remove the `React.useEffect` in `WagmiConfig` and call the `autoConnect` function inside the client (e.g. constructor) instead. This way `createClient({ autoConnect: true })` in the Vanilla JS version would work without calling separately.

Haven't spent the time yet to figure out the best way to do this reliably across different rendering environments though (React versus Vanilla, client versus SSR).